pkg/operation/botanist/component/istio/istiod.go

```diff
@@ -92,13 +94,24 @@ type Values struct {
 	NamePrefix string
 }
 
+// Interface contains functions for an Istio deployer.
+type Interface interface {
+	component.DeployWaiter
+	// AddIngressGateway adds another ingress gateway to the existing Istio deployer.
+	AddIngressGateway(values IngressGatewayValues)
+	// GetValues returns the configured values of the Istio deployer.
+	GetValues() Values
+}
+
+var _ Interface = (*istiod)(nil)
```

Comment:
It's a helper to get compiler errors in case `istiod` doesn't implement the `Interface` interface. We use this technique at various places, e.g. [here](https://github.com/gardener/gardener/blob/0b5cdc002b481e77e94d6fae2a1dc9daf1a3a550/pkg/utils/secrets/manager/manager.go#L112).